src/Grpc.Net.Client/Internal/GrpcProtocolConstants.cs

```diff
@@ -86,14 +88,24 @@ internal static string GetMessageAcceptEncoding(Dictionary<string, ICompressionP
         }
 
         static GrpcProtocolConstants()
+        {
+            UserAgentHeader = "User-Agent";
+            UserAgentHeaderValue = GetUserAgentString();
+            TEHeader = "TE";
+            TEHeaderValue = "trailers";
+
+            DefaultMessageAcceptEncodingValue = GetMessageAcceptEncoding(DefaultCompressionProviders);
+        }
+
+        private static string GetUserAgentString()
```

Comment:
nit: extract the functionality for computing the user agent string to a separate  static class (and then add tests for that class)?